core/ide/che-core-ide-app/src/main/java/org/eclipse/che/ide/context/AppContextImpl.java

```diff
@@ -175,6 +177,21 @@ public void setProjectsRoot(String projectsRoot) {
     }
 
     @Override
+    public String getWsAgentURL() {
+        return wsAgentURL;
```

Comment:
I think it is not needed. you can use context.getDevMachine().getWsAgentBaseUrl() instead